akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -499,12 +517,28 @@ class ShardRegion(
               log.debug("Request shard [{}] home", shardId)
               coordinator.foreach(_ ! GetShardHome(shardId))
             }
-            if (totalBufferSize >= bufferSize) {
-              log.debug("Buffer is full, dropping message for shard [{}]", shardId)
+            val totBufSize = totalBufferSize
+            if (totBufSize >= bufferSize) {
+              if (loggedFullBufferWarning)
+                log.debug("Buffer is full, dropping message for shard [{}]", shardId)
+              else {
+                log.warning("Buffer is full, dropping message for shard [{}]", shardId)
+                loggedFullBufferWarning = true
+              }
               context.system.deadLetters ! msg
             } else {
               val buf = shardBuffers.getOrElse(shardId, Vector.empty)
               shardBuffers = shardBuffers.updated(shardId, buf :+ ((msg, snd)))
+
+              // log some insight to how buffers are filled up every 10% of the buffer capacity
+              val tot = totBufSize + 1
+              if (tot % (bufferSize / 10) == 0) {
+                val logMsg = s"ShardRegion for [$typeName] is using [${100.0 * tot / bufferSize} %] of its buffer capacity."
+                if (tot <= bufferSize / 2)
+                  log.info(logMsg)
```

Comment:
Yeah, I got that. Not sure how noisy it will be (if buf fluctuates a lot around 20% for example).
Let's go with INFO - we can always tune down if someone complains